x-pack/plugins/fleet/server/services/package_policy.ts

```diff
@@ -236,25 +236,37 @@ class PackagePolicyService implements PackagePolicyServiceInterface {
       force?: true;
     }
   ): Promise<PackagePolicy[]> {
-    const agentPolicyIds = new Set(packagePolicies.map((pkgPol) => pkgPol.policy_id));
+    const agentPolicyIds = new Set(packagePolicies.map((pkgPolicy) => pkgPolicy.policy_id));
 
     for (const agentPolicyId of agentPolicyIds) {
       await validateIsNotHostedPolicy(soClient, agentPolicyId, options?.force);
     }
 
+    const packageInfos = await getPackageInfoForPackagePolicies(packagePolicies, soClient);
+
     const isoDate = new Date().toISOString();
     // eslint-disable-next-line @typescript-eslint/naming-convention
     const { saved_objects } = await soClient.bulkCreate<PackagePolicySOAttributes>(
-      packagePolicies.map((packagePolicy) => {
+      await pMap(packagePolicies, async (packagePolicy) => {
         const packagePolicyId = packagePolicy.id ?? uuid.v4();
         const agentPolicyId = packagePolicy.policy_id;
 
-        const inputs = packagePolicy.inputs.map((input) =>
+        let inputs = packagePolicy.inputs.map((input) =>
           assignStreamIdToInput(packagePolicyId, input)
         );
 
         const { id, ...pkgPolicyWithoutId } = packagePolicy;
 
+        if (packagePolicy.package) {
+          const pkgInfo = packageInfos.get(
+            `${packagePolicy.package.name}-${packagePolicy.package.version}`
+          );
+
+          inputs = pkgInfo
+            ? await _compilePackagePolicyInputs(pkgInfo, packagePolicy.vars || {}, inputs)
+            : inputs;
+        }
+
```

Comment:
could you add the `elasticsearch` part too that we have in the other create method? 
```
elasticsearch = pkgInfo.elasticsearch;
``